tree-construction/template.dat

```diff
@@ -1610,14 +1613,18 @@ eof table
 #data
 <template><form><input name="q"></form><div>second</div></template>
 #errors
+no doctype
 #document
-| <template>
-|   content
-|     <form>
-|       <input>
-|         name="q"
-|     <div>
-|       "second"
+| <html>
+|   <head>
+|     <template>
+|       content
+|         <form>
+|           <input>
+|             name="q"
+|         <div>
+|           "second"
+|   <body>
```

Comment:
see also prior discussion in https://github.com/html5lib/html5lib-tests/pull/161 w. @annevk

This was motivated by the fact that all browsers get, with `#document-fragment: template`:

```
|<template>
| content
|   <template>
|     content
|       <form>
|         <input>
|           name="q"
|       <div>
|         "second"
```

Given, AFAIK, this test finds an underlying difference between how html5lib-tests is defined (the output of the HTML parser) and what `innerHTML` does in browsers.

I think WPT's `/html/syntax/parsing/test.js` needs to handle https://w3c.github.io/DOM-Parsing/#the-innerhtml-mixin changing the context node:

> If the context object is a template element, then let context object be the template's template contents (a DocumentFragment).

And indeed, adding a breakpoint and manually running this:

```
> console.log(test_serializer(container_elem.content))
[Log] #document-fragment
| <template>
|   content
|     <form>
|       <input>
|         name="q"
|     <div>
|       "second"
```

Reply:
https://github.com/web-platform-tests/wpt/pull/39474 fixes the WPT html5lib-tests runner.

Reply:
Let's restore `#document-fragment` here then.

Reply:
Updated!